### PR TITLE
Update API documentation to support APIv4 information

### PR DESCRIPTION
**Pull request by seamuslee001**

### CODE REVIEW

File: docs/api/index.md [hunk 1]
```diff
@@ -19,15 +19,27 @@ CiviCRM, using the API explorer and the API parameter list.
 
 For help creating your own API custom calls, see [civix generate:api](/extensions/civix.md#generate-api)
 
+## Versions
+
+CiviCRM has two different versions of the API. Version 3 is considered the current stable version and Version 4 is the currently worked on version.
```
Comment (totten) [inline docs/api/index.md]: I don't think this assertion creates quite the right impression and will quickly grow stale. If you think some kind of comparison/trade-off is really needed, then maybe point out that (a) v3 has been around longer and has more entities (b) v4 has cleaner/more consistent design.

File: docs/api/index.md [hunk 2]
```diff
@@ -54,3 +68,30 @@ From CiviCRM v5.8 the APIv3 explorer will now be able to show examples that are
 ## Changelog
 
 All important changes made to the API are recorded in [API changes](/api/changes.md).
+
+## Version 4 Changes from version 3
```
Comment (totten) [inline docs/api/index.md]: Use Consistent case?

File: docs/api/index.md [hunk 3]
```diff
@@ -40,12 +52,14 @@ have been testing.
...
 action or complete your create or update action.
 
 ## API examples
 
-From the API explorer, you can click on the **Examples** tab to find examples of API calls which are based on automated tests within the source code. You can also [explore these examples on GitHub](https://github.com/civicrm/civicrm-core/tree/master/api/v3/examples).
+For API version 3 from the API explorer, you can click on the **Examples** tab to find examples of API calls which are based on automated tests within the source code. You can also [explore these examples on GitHub](https://github.com/civicrm/civicrm-core/tree/master/api/v3/examples).
+
+For API version 4 the explorer doesn't explose any examples but does show off code options.
 
 ### API Examples in your extensions
```
Comment (totten) [inline docs/api/index.md]: Use consistent Case?

File: docs/extensions/civix.md [hunk 4]
```diff
@@ -165,6 +165,8 @@ If you want your extension to store data in the database, then you will need to
     
 Now your entity should be ready to use. Try it out with `cv api MyEntity.create` and `cv api MyEntity.get`. Then [add some tests](#generate-test).
 
+By default when you generate an entity you will be genering an API version 4 file. To generate an API version 3 interface you need to specifiy `--api-version 3,4` or just `--api-version 3` 
+
```
Comment (totten) [inline docs/extensions/civix.md]: Good catch

File: docs/api/index.md (hunk 2 above)
Comment (MikeyMJCO) [reply docs/api/index.md]:
```suggestion
## Changes from APIv3
```

File: docs/api/usage.md [hunk 5]
```diff
@@ -117,10 +117,61 @@ The object-oriented client can connect to a local or remote CiviCRM
 instance. For details about connection parameters, see the docblock in
 [class.api.php](https://github.com/civicrm/civicrm-core/blob/master/api/class.api.php).
 
+## PHP API version 4
+
+In API version 4 there are 2 distinct formats for querying the API, firstly an Object Oriented format
+
+```php
+$aCLs = \Civi\Api4\ACL::get()
```
Comment (totten) [inline docs/api/usage.md]: `$acls` and `$acl`?

File: docs/api/index.md (hunk 3 above)
Comment (MikeyMJCO) [reply docs/api/index.md]:
```suggestion
For APIv3 from the API explorer, you can click on the **Examples** tab to find examples of API calls which are based on automated tests within the source code. You can also [explore these examples on GitHub](https://github.com/civicrm/civicrm-core/tree/master/api/v3/examples).

For APIv4 the explorer does not show any examples but does show off code options. Adding examples to this area of the UI would be beneficial.
```

File: docs/extensions/civix.md (hunk 4 above)
Comment (MikeyMJCO) [reply docs/extensions/civix.md]:
```suggestion
By default when you generate an entity you will be generating an APIv4 entity. To generate an APIv3 (or both) interface you need to specify `--api-version 3,4` or just `--api-version 3`
```

**Comment by MikeyMJCO** [comment]
@seamuslee001 is this stale?

**Comment by seamuslee001** [comment]
I would say yes

**Comment by MikeyMJCO** [comment]
Can I close?

**Comment by seamuslee001** [comment]
Yes

**Pull request closed without merging**